client/client-multi/client-c/src/kaa/kaa_notification_manager.c

```diff
@@ -0,0 +1,1001 @@
+                kaa_list_t *notifications = NULL;
+                kaa_notification_wrapper_t *notification = NULL;
+                uint16_t uid_length = 0;
+                uint16_t notifications_count = KAA_NTOHS(*((uint16_t *) reader->current));
+                reader->current += sizeof(uint16_t);
+                extension_length -= sizeof(uint16_t);
+                KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Notifications count is %u", notifications_count);
+                if (notifications_count) {
+                    while (notifications_count--) {
+                        notification = (kaa_notification_wrapper_t *) KAA_MALLOC(sizeof(kaa_notification_wrapper_t));
+                        KAA_RETURN_IF_NIL(notification, KAA_ERR_NOMEM);
+                        notification->seq_num = KAA_NTOHL(*((uint32_t *) reader->current));
+                        reader->current += sizeof(uint32_t);
+                        extension_length -= sizeof(uint32_t);
+                        notification->type = (*((uint8_t *) reader->current) == SYSTEM)? SYSTEM : CUSTOM;
+                        reader->current += sizeof(uint16_t);
+                        extension_length -= sizeof(uint16_t);
+                        uid_length = KAA_NTOHS(*((uint16_t *) reader->current));
+                        reader->current += sizeof(uint16_t);
+                        extension_length -= sizeof(uint16_t);
+                        notification->size = KAA_NTOHL(*((uint32_t *) reader->current));
+                        reader->current += sizeof(uint32_t);
+                        extension_length -= sizeof(uint32_t);
+                        notification->topic_id =  KAA_NTOHLL(*((uint64_t *) reader->current));
+                        reader->current += sizeof(uint64_t);
+                        extension_length -= sizeof(uint64_t);
+                        KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Topic id is %u", notification->topic_id);
+                        if (uid_length) {
+                            kaa_notifications_uid_t *uid = (kaa_notifications_uid_t *) KAA_MALLOC(sizeof(kaa_notifications_uid_t));
+                            KAA_RETURN_IF_NIL(uid, KAA_ERR_NOMEM);
+                            uid->length = uid_length;
+                            uid->data = KAA_MALLOC(sizeof(uid_length));
+                            err = kaa_platform_message_read_aligned(reader, uid->data, uid->length);
+                            if (err) {
+                                KAA_LOG_WARN(self->logger, KAA_ERR_BADDATA, "Failed to read UID body");
+                            }
+                            extension_length -= kaa_aligned_size_get(uid->length);
+                            if (!self->uids) {
+                                KAA_RETURN_IF_NIL(self->uids = kaa_list_create(uid), KAA_ERR_NOMEM);
+                            } else {
+                                KAA_RETURN_IF_NIL(kaa_list_push_back(self->uids, uid), KAA_ERR_NOMEM);
```

Comment:
Please, review all kaa_list_push_back() calls, maybe it's better to push into the head.